### PR TITLE
update ocf-tv to work without avahi

### PR DESCRIPTION
**Pull request by abizer**
also slightly better argparse and reorganize the script a bit

+1 to @matthew-mcallister for https://github.com/ocf/puppet/pull/195 for which this is the counterpart.

### CODE REVIEW

File: staff/lab/ocf-tv [hunk 1]
```diff
@@ -50,91 +41,95 @@ def wait_for_port(host, port, timeout=5, ssh_proc=None):
...
+            close_audio_tunnel()
+
+        # set up an explicit tunnel sink that points to the local end of our SSH tunnel
+        # if this succeeds, it should return a loaded module index, e.g. '25' that
+        # you can see by running `pactl list modules`
+        ret = subprocess.call(('pactl', 'load-module', 'module-tunnel-sink-new',
+                               'server=tcp:localhost:{}'.format(local_port),
+                               'sink_name=' + TUNNEL_NAME), stdout=subprocess.DEVNULL)
+        
+        ret = subprocess.call(('pactl', 'set-default-sink', TUNNEL_NAME))
```
Comment (matthew-mcallister) [inline staff/lab/ocf-tv]: This can fail right? So shouldn't this be `subprocess.check_call`?

File: staff/lab/ocf-tv [hunk 2]
```diff
@@ -50,91 +41,95 @@ def wait_for_port(host, port, timeout=5, ssh_proc=None):
...
+
+        # move everything that's currently playing to the new sink
+        cmd = "pacmd list-sink-inputs \
+              | awk '/index: / { print $2 }' \
+              | xargs -I {} pacmd move-sink-input {} " + TUNNEL_NAME
+                              
+        subprocess.call(cmd, shell=True)
```
Comment (matthew-mcallister) [inline staff/lab/ocf-tv]: Ditto about `check_call`?

File: staff/lab/ocf-tv [hunk 3]
```diff
@@ -50,91 +41,95 @@ def wait_for_port(host, port, timeout=5, ssh_proc=None):
...
+                              
+    except Exception as e:
+        print(e)
+        exit(1)
+
+def close_audio_tunnel():
+    return subprocess.call(('pactl', 'unload-module', 'module-tunnel-sink-new'),
+                           stdout=subprocess.DEVNULL)
```
Comment (matthew-mcallister) [inline staff/lab/ocf-tv]: It shouldn't be a problem, but will this fix up things if the module gets loaded twice?

File: staff/lab/ocf-tv [hunk 4]
```diff
@@ -50,91 +41,95 @@ def wait_for_port(host, port, timeout=5, ssh_proc=None):
...
-    try:
-        sink_name = subprocess.check_output("pactl list sinks | awk '/Name: tunnel/ { print $2 }'", shell=True)
-        return str(sink_name.strip(), 'ascii')
-    except:
-        print('Failed to get the sink for the tunnel. Did you start it?')
-        sys.exit(1)
+def check_volume(val):
+    x = int(val)
+    if abs(x) > 150:
+        raise argparse.ArgumentTypeError('Volume out of bounds: {} not in [0, 150]'.format(val))
```
Comment (matthew-mcallister) [inline staff/lab/ocf-tv]: Should this read `[0, 150]` or `[-150, 150]`?

File: staff/lab/ocf-tv [hunk 5]
```diff
@@ -143,22 +138,44 @@ if __name__ == '__main__':
...
+        help='volume in percent between 0% and 150%.'
+             ' +/- a relative value is also allowed.'
+    )
+    subparser_volume.set_defaults(func=volume)
+    
+    subparser_mute = command_subparser.add_parser(
+        'mute',
+        help="Toggle mute on the TV"
+    )
+    subparser_mute.set_defaults(func=volume)
```
Comment (matthew-mcallister) [inline staff/lab/ocf-tv]: Nit: trailing commas in this section.

File: staff/lab/ocf-tv [hunk 6]
```diff
@@ -50,91 +41,95 @@ def wait_for_port(host, port, timeout=5, ssh_proc=None):
...
+        # move everything that's currently playing to the new sink
+        cmd = "pacmd list-sink-inputs \
+              | awk '/index: / { print $2 }' \
+              | xargs -I {} pacmd move-sink-input {} " + TUNNEL_NAME
+                              
+        subprocess.call(cmd, shell=True)
+                              
+    except Exception as e:
+        print(e)
+        exit(1)
```
Comment (matthew-mcallister) [inline staff/lab/ocf-tv]: How is this different from not catching? Unless you want to hide the traceback, that is.

File: staff/lab/ocf-tv (hunk 3 above)
Comment (tahabi) [reply staff/lab/ocf-tv]: yes - if you load module-tunnel-sink-new twice, the second time it gets loaded with a new sink name. unloading the module unloads both.

**Pull request merged**